packages/core/src/store/anchor-request-store.ts

```diff
@@ -98,12 +106,22 @@ export class AnchorRequestStore extends ObjectStore<StreamID, AnchorRequestData>
     let numEntries = 0
     do {
       try {
-        const batch = await this.store.find({
+        let timeout
+        const timeoutPromise = new Promise<null>((resolve) => {
+          timeout = setTimeout(() => {
+            this.#logger.warn(`Timed out while waiting for AnchorRequestStore to fetch a batch`)
+            resolve(null)
+          }, this.#infiniteListBatchTimeoutMs)
+        })
+
+        const batchPromise = this.store.find({
           limit: batchSize,
           useCaseName: this.useCaseName,
           gt: generateKey(gt),
         })
-        if (batch.length > 0) {
+        const batch = await Promise.race([batchPromise, timeoutPromise])
+        clearTimeout(timeout)
```

Comment:
Adding a log here as well after this line to indicate the completion of the find.

``` 
 if (batch !== null) {
  this.#logger.debug(`Completed 'find' operation with batchSize: ${batchSize}, retrieved ${batch.length} items`);
}
```